docs/getting_started.md

```diff
@@ -0,0 +1,246 @@
+
+```python
+my_app.commands.execute_command('open')
+# prints "open file!" from the `open_file` function registered above.
+```
+
+### Command Arguments and Dependency Injection
+
+The `execute_command` function does accept `*args` and `**kwargs` that will
+be passed to the command.  However, very often in a GUI application
+you may wish to infer some of the arguments from the current state of the
+application.  For example, if you have menu item linked to a "close window",
+you likely want to close the current window. For this, `app-model` uses
+a dependency injection pattern, provided by the
+[`in-n-out`](https://github.com/pyapp-kit/in-n-out) library.
+
+The application has a [`injection_store`][app_model.Application.injection_store]
+attribute that is an instance of an `in_n_out.Store`.  A `Store` is a collection
+of "providers" (functions that can be called to return an instance of a given
+type), and "processors" (functions that accept an instance of a given type and
+do something with it). The `Store` is used to provide the arguments to the
+command when it is executed, based on the type annotations in the command
+function definition.
+
+Here's an example.  Let's say an application has a `User` object with a `name()`
+method:
+
+```python
+class User:
+    def name(self):
+        return 'John Doe'
+```
+
+Assume the application has some way of retrieving the current user:
+
+```python
+def get_current_user() -> User:
+    # ... get the current user from somewhere
+    return User()
+```
+
+We register this provider function with the application's injection store:
+
+```python
+my_app.injection_store.register_provider(get_current_user)
+```
+
+Now commands may be defined that accept a `User` argument, and used
+for callbacks in actions registered with the application.
+
+```python
+def print_user_name(user: User) -> None:
+    print(f"Hi {user.name()}!")
+
+action = Action(
+    id='greet',
+    title="Greet Current User",
+    callback=print_user_name,
+)
+
+my_app.register_action(action)
+my_app.commands.execute_command('greet')
+# prints "Hi John Doe!"
+```
+
+## Connecting a GUI framework
+
+Of course, most of this is useless without some way to connect the application
+to a GUI framework.  The [`app_model.backends`][app_model.backends] module
+provides functions that map the `app-model` model onto various GUI framework models.
+
+!!! note "erm... someday 😂"
+
+    Well, really it's just Qt for now, but the abstraction is valuable for the
+    ability to swap backends.  And we hope to add more backends if the demand is
+    there.
+
+### Qt
+
+Currently, we don't have a generic abstraction for the application window, so
+users are encouraged to directly use the classes in the `app_model.backends.qt`
+module.  One of the main classes is the [`QModelMainWindow`][app_model.backends.qt.QModelMainWindow] object: a subclass of `QMainWindow` that knows how to map
```

Comment:
Biased but might be nice to add a bit about `QModelMenu` ?

Reply:
I agree, more of the objects in the qt backend module would benefit from examples.  I think maybe that warrants another full page dedicated to examples using Qt.  Will do in a follow up PR.

Reply:
#144